bigbluebutton/greenlight
# Created the base User Model

closes #3212 

## Description
1. Removed unneeded test folder
2. Added faker, factory_bot, shoulda_matchers to help with testing
3. Added a migration and model for `User`
4. Created a factory and added tests for User

## Testing Steps
1. Pull the changes
2. Open the rails console
3. Try creating a `User` with a variety of options to make sure everything works as expected
```
User.create(name: "...", email: "...", ...)
```
(Try leaving some as nil as make sure the errors are correct)
